### PR TITLE
NMS-13117: Added smoke test to verify the reloading of daemons

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-13117

### CODE REVIEW

**Comment by cgorantla** [comment]
Looks good to me. 
Can you consider extending [DaemonReloadEnum](https://github.com/OpenNMS/opennms/blob/6afe2c7a121d32687765c2ff827dbdc8b15ca62f/core/lib/src/main/java/org/opennms/core/daemon/DaemonReloadEnum.java#L31) to include all missing Daemons like Alarmd/Collectd ?

Then I would opt for different Enum in Tests to pick and choose which daemons to run in smoke test.

**Comment by christianpape** [comment]
> Looks good to me.
> Can you consider extending [DaemonReloadEnum](https://github.com/OpenNMS/opennms/blob/6afe2c7a121d32687765c2ff827dbdc8b15ca62f/core/lib/src/main/java/org/opennms/core/daemon/DaemonReloadEnum.java#L31) to include all missing Daemons like Alarmd/Collectd ?
> 
> Then I would opt for different Enum in Tests to pick and choose which daemons to run in smoke test.

Ok, I’ll add Alarm and Collectd to the enum, but do we really need a different enum for the smoke-test? All daemons except Telemetryd reload instantly. I think Telemetryd is important to test. So, testing all does not really add much delay to the test...

File: core/lib/src/main/java/org/opennms/core/daemon/DaemonReloadEnum.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
 public enum DaemonReloadEnum {
 
+    ALARMD("Alarmd"),
```
Comment (cgorantla) [inline core/lib/src/main/java/org/opennms/core/daemon/DaemonReloadEnum.java:33]: Looks like the names for these Daemons are in lower case `alarmd` and `collectd`.

Comment (christianpape) [reply core/lib/src/main/java/org/opennms/core/daemon/DaemonReloadEnum.java]: You're right

**Comment by cgorantla** [comment]
Tests are failing.  May be One of `collectd` or `alarmd` reload daemon is broken ?

**Comment by christianpape** [comment]
> Tests are failing. May be One of `collectd` or `alarmd` reload daemon is broken ?

Fixed.

**Comment by cgorantla** [comment]
> > Tests are failing. May be One of `collectd` or `alarmd` reload daemon is broken ?
> 
> Fixed.

Okay, got it. [This](https://github.com/OpenNMS/opennms/blob/d716e549b975c7fec4bcf25e399c7fd381963a47/opennms-services/src/main/java/org/opennms/netmgt/collectd/Collectd.java#L1120) forces it to `Collectd` instead of name.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**